packages/api-client/src/client.ts

```diff
@@ -94,25 +94,34 @@ export default class ApiClient {
       responseBody = {};
     }
     if (response.ok) {
-      return responseBody;
+      return { body: responseBody, headers: response.headers };
     } else {
       throw new ApiClientError(request, response, responseBody);
     }
   }
 
   async get<Response>(request: ApiClientGetOptions): Promise<Response> {
-    return await this.request({ ...request, method: 'GET' });
+    const response = await this.request({ ...request, method: 'GET' });
+    return response.body;
   }
 
-  async post<Response>(request: ApiClientPostOptions): Promise<Response> {
-    return await this.request({ ...request, method: 'POST' });
+  async post<Response extends { location?: string }>(
+    request: ApiClientPostOptions
+  ): Promise<Response> {
+    const response = await this.request({ ...request, method: 'POST' });
+    return {
+      ...response.body,
+      location: response.headers.get('Location'),
```

Comment:
I modified the post request so that we can read the location header to get the id for created resources. Turns out I didn't actually need it, but should be useful in the future.